UnchainedLauncherCore/src/API/ServerBrowser.cs

```diff
@@ -0,0 +1,248 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Net;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using UnchainedLauncher.Core.API;
+using UnchainedLauncher.Core.JsonModels.Metadata.V3;
+using System.Runtime.Loader;
+using PropertyChanged;
+
+namespace UnchainedLauncher.Core.API
+{
+    public record Ports(
+        int Game,
+        int Ping,
+        int A2s
+    );
+    [AddINotifyPropertyChangedInterface]
+    public record C2ServerInfo {
+        public bool PasswordProtected { get; set; } = false;
+        public string Name { get; set; } = "";
+        public string Description { get; set; } = "";
+        public Ports Ports { get; set; } = new(7777, 3075, 7071);
+        // TODO: The selection of the Mod datatype here is potentially
+        // incorrect. Change it to whatever is easier and works with the backend
+        public Mod[] Mods { get; set; } = Array.Empty<Mod>(); 
+    };
+    // TODO: This part of the API should be stabilized
+    // as-is, there are numerous different kinds of "server" objects depending
+    // on where they're coming from/where they're going
+    public record ServerInfo : C2ServerInfo{
+        public String CurrentMap { get; set; } = "";
+        public int PlayerCount { get; set; } = 0;
+        public int MaxPlayers { get; set; } = 0;
+        public ServerInfo() { }
+        public ServerInfo(C2ServerInfo info, A2S_INFO a2s) : base(info) {
+            Update(a2s);
+        }
+        public bool Update(A2S_INFO a2s)
+        {
+            bool wasChanged = (
+                MaxPlayers != a2s.MaxPlayers
+                || PlayerCount != a2s.Players
+                || CurrentMap != a2s.Map);
```

Comment:
I think this comparison can be simplified by constructing a tuple and comparing them:
```c#
(MaxPlayers, PlayerCount, CurrentMap) != (a2s.MaxPlayers, a2s.Players, a2s.Map)
```